Google.Api.Gax/TaskExtensions.cs

```diff
@@ -49,7 +50,98 @@ public static void WaitWithUnwrappedExceptions(this Task task)
                 // Using ExceptionDispatchInfo to keep the original exception stack trace.
                 ExceptionDispatchInfo.Capture(e.InnerExceptions.FirstOrDefault() ?? e).Throw();
             }
+        }
+
+        /// <summary>
+        /// Synchronously waits for the given task to complete.
+        /// Any <see cref="AggregateException"/> thrown is unwrapped to the first inner exception.
+        /// </summary>
+        /// <param name="task">The task to wait for.</param>
+        /// <param name="timeout">A TimeSpan that represents the number of milliseconds to wait, or
+        /// -1 milliseconds to wait indefinitely.</param>
+        public static void WaitWithUnwrappedExceptions(this Task task, TimeSpan timeout)
+        {
+            try
+            {
+                task.Wait(timeout);
+            }
+            catch (AggregateException e)
+            {
+                // Unwrap the first exception, a bit like await would.
+                // It's very unlikely that we'd ever see an AggregateException without an inner exceptions,
+                // but let's handle it relatively gracefully.
+                // Using ExceptionDispatchInfo to keep the original exception stack trace.
+                ExceptionDispatchInfo.Capture(e.InnerExceptions.FirstOrDefault() ?? e).Throw();
+            }
+        }
 
+        /// <summary>
+        /// Synchronously waits for the given task to complete.
+        /// Any <see cref="AggregateException"/> thrown is unwrapped to the first inner exception.
+        /// </summary>
+        /// <param name="task">The task to wait for.</param>
+        /// <param name="millisecondsTimeout">The number of milliseconds to wait, or
+        /// -1 to wait indefinitely.</param>
+        public static void WaitWithUnwrappedExceptions(this Task task, int millisecondsTimeout)
```

Comment:
Do we use these overloads with `int` timeout parameters? If not, I'd rather we didn't include them here, as I don't really like `int`s being used as durations.

Reply:
Hmm. The point would be to make it trivial to go from a `Wait(anything)` to `WaitWithUnwrappedExtensions(anything)`, but I do see your point. Let's see what Amanda thinks :)

Reply:
Note that if we do remove the milliseconds-based ones, we'd want to add a "TimeSpan and CancellationToken" one - which doesn't exist on Wait itself.

Reply:
Agree with `int` as duration being ugly, but it's already in `Wait`, and we are just wrapping `Wait` so, might make sense here. These are just helper methods after all. I'm more incline to leave it as is.